src/main/java/com/ibanity/apis/client/products/xs2a/services/impl/SynchronizationServiceImpl.java

```diff
@@ -36,21 +57,34 @@ public Synchronization create(SynchronizationCreationQuery synchronizationCreati
                 .customerOnline(synchronizationCreationQuery.getCustomerOnline())
                 .customerIpAddress(synchronizationCreationQuery.getCustomerIpAddress())
                 .build();
-        String url = getUrl();
+        String url = getUrl(null, null, null);
         RequestApiModel request = buildRequest(Synchronization.RESOURCE_TYPE, synchronization);
         HttpResponse response = ibanityHttpClient.post(buildUri(url), request, synchronizationCreationQuery.getAdditionalHeaders(), synchronizationCreationQuery.getCustomerAccessToken());
         return mapResource(response, (SynchronizationMapper::map));
     }
 
     @Override
     public Synchronization find(SynchronizationReadQuery synchronizationReadQuery) {
-        String url = getUrl() + "/" + synchronizationReadQuery.getSynchronizationId();
+        String url = getUrl(null, null, synchronizationReadQuery.getSynchronizationId())
+                + "/"
+                + synchronizationReadQuery.getSynchronizationId().toString();
         HttpResponse response = ibanityHttpClient.get(buildUri(url), synchronizationReadQuery.getAdditionalHeaders(), synchronizationReadQuery.getCustomerAccessToken());
         return mapResource(response, (SynchronizationMapper::map));
     }
 
-    private String getUrl() {
-        String url = apiUrlProvider.find(IbanityProduct.Xs2a, "customer", "synchronizations");
+    private String getUrl(UUID financialInstitutionId, UUID accountInformationAccessRequestsId, UUID synchronizationId) {
```

Comment:
I think the code would be more clear if you create 2 getUrl methods:

getUrl()
getUrl(UUID financialInstitutionId, UUID accountInformationAccessRequest)

Doing that will remove a couple of if statements.